binding/java/src/ai/picovoice/porcupine/exception/PorcupineException.java

```diff
@@ -11,11 +11,44 @@
 package ai.picovoice.porcupine;
 
 public class PorcupineException extends Exception {
+    private final String message;
+    private final String[] messageStack;
+
     public PorcupineException(Throwable cause) {
         super(cause);
+        this.message = cause.getMessage();
+        this.messageStack = null;
     }
 
     public PorcupineException(String message) {
         super(message);
+        this.message = message;
+        this.messageStack = null;
+    }
+
+    public PorcupineException(String message, String[] messageStack) {
+        super(message);
+        this.message = message;
+        this.messageStack = messageStack;
+    }
+
+    public String[] getMessageStack() {
+        return this.messageStack;
+    }
+
+    @Override
+    public String getMessage() {
+        StringBuilder sb = new StringBuilder(message);
+        if (messageStack != null) {
+            if (messageStack.length > 0) {
+                sb.append(":");
+                for (int i = 0; i < messageStack.length; i++) {
+                    sb.append(String.format("\n  [%d] %s", i, messageStack[i]));
+                }
+            } else {
+                sb.append(".");
+            }
+        }
```

Comment:
I don't think placing the period is necessary
```suggestion
        if (messageStack != null) {
            if (messageStack.length > 0) {
                sb.append(":");
                for (int i = 0; i < messageStack.length; i++) {
                    sb.append(String.format("\n  [%d] %s", i, messageStack[i]));
                }
            }
        }
```

Reply:
I added period everywhere, those the prs I merged have the period?

Reply:
oh ok, weird I put it on java only haha